src/main/kotlin/org/flaxo/plagiarism/graph2viz.kt

```diff
@@ -34,6 +35,14 @@ import kotlin.random.Random
 fun Graph.toViz(canvasWidth: Int = 800, canvasHeight: Int = 500): Viz =
         toViz(canvasWidth.toDouble(), canvasHeight.toDouble())
 
+private const val defaultArrowLengthX = 30.0
+private const val defaultArrowLengthY = 8.0
+
+private val lines = mutableListOf<Line>()
```

Comment:
The following lists `lines`, `arrowLines`, `circles`, `texts` are defined in the file's root which makes them static. I suppose that it is not a best practice to do so. Probably, the lists should be extracted to a single class e.g. `GraphState` which instance will be created and used inside the `Graph.toViz` function.

Reply:
As I understand, there is no essential difference between static and instance members when they're compiled to js, so there is no point in trying to make them nonstatic. But firstly I moved these entities in the collections because I could not separate new `Line` objects that are now used as the arrow lines from the `Line` objects that represent the links.

Reply:
Probably, I wasn't quite clear. `GraphState` instance should be just a local variable. Therefore there is a difference)

The problem is that the root declaration allows to create only a single instance of graph vizualization because it uses mutable static state. The introduction of `GraphState` solves the issue.

```kotlin
class GraphState(
    val lines: List<Line>,
    val circles: List<Circle>,
    ...
)

fun Graph.toViz(canvasWidth: Double = 800.0, canvasHeight: Double = 500.0): Viz = viz {
    ...
    val state = GraphState(lines=lines, circles=circles, ...)
    ...
}
```

Reply:
Yes, in the root declaration may be only one state. I suppose there cannot be multiple instances of the graph state. And you miss the point that the `GraphState` cannot be only a local variable in the `Graph.toViz` function because the shape lists also used in the `Viz.refreshGraph` function.

Reply:
I think that this is a topic for another issue, so I suggest to return the previous state of this file and to use `all<Line>()`-like functions to manipulate the shape objects

Reply:
Several graphs representation may be a feature in some future release. For example, analyses for several course tasks can be shown at the same screen or even the same canvas. Therefor we should not ignore the possibility :)

And about the `GraphState` local variable. Correct me if I'm wrong but you can pass it as an argument to `Viz.refreshGraph` function.

Reply:
Such a possibility is the reason why it is better for you to deal with this case. Please look at the current state of the file.